scripts/inference.py

```diff
@@ -355,9 +373,53 @@
     # [ATTENTION] for GPTQ on AIU, we must always instantiate an unfused
     # model, the adapter will take care of converting key/values from
     # ckpt into the appropriate form for the model
-    if fused_weights:
+    if fused_weights and is_aiu_backend:
         raise ValueError("GPTQ checkpoints on AIU must always run with --unfuse_weights")
     default_dtype = None  # GPTQ dtype always comes from ckpt, can't be enforced
+elif args.quantization == "int8":
+    def select_int8_module(
+        module_name: str | None = None,
+        smoothquant: bool = True,
+        smoothquant_layers: list[str] | None = None,
+    ):
+        if module_name is None:
+            return "int8_aiu"
+        smoothquant_on_module = (
+            any([m in module_name for m in smoothquant_layers])
+            if smoothquant_layers is not None
+            else True
+        )
+        use_smoothquant = smoothquant and smoothquant_on_module
+        return "int8_smoothquant_aiu" if use_smoothquant else "int8_aiu"
+
+    if args.int8_smoothquant:
+        # TODO: consider saving this info into config during quantization
+        if any("granite" in p for p in [args.model_path, args.architecture]):
```

Comment:
will this work for things like Granite or Roberta instead of granite and roberta? some of the HF checkpoint names have capitalized letters